spec/acceptance/default_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper_acceptance'
+
+describe 'nftables class' do
+  context 'configure default nftables service' do
+    it 'works idempotently with no errors' do
+      pp = <<-EOS
+      # default mask of firewalld service fails if service not installed.
```

Comment:
That sounds like a bug in the module.

Reply:
It's bug in puppet possibly.

```puppet
service{'foo':
  enable => false,
}
```
is okay if service foo does not exist however with `enable => mask` it gives an error.

So you want to do something different depending upon if firewalld is installed or not but this module
does not  enforce if firewalld should be installed.

Puppet should mask anyway.

```
systemctl mask foobar.service
Unit foobar.service does not exist, proceeding anyway.
Created symlink /etc/systemd/system/foobar.service → /dev/null.
```
seems to work.

Will see if that is upstream.

Reply:
Nothing exact - https://tickets.puppetlabs.com/browse/PUP-10814 submitted.

So given the behviour of puppet mask is sensible default in case the package is installed which is normally the case but for other cases like this one it needs an option.

Resolved for me.

Reply:
I always like to add a link to the issue as a code comment because that would answer my question. Other that it looks good to me.